### PR TITLE
Feature/pharmacist model

### PR DESCRIPTION
**Pull request by muhammad-nabih**

### CODE REVIEW

File: database/models/pharmacist.model.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        required: [true, 'license number is required'],
+        unique: true,
+        maxLength: 50
+    },
+    username: {
+        type: String,
+        required: [true, 'Username is required'],
+        unique: true,
+        minLength: [3, 'Username is too short'],
+        maxLength: 50
```
Comment (mennaseif) [inline database/models/pharmacist.model.ts]: image : {
            type: String,
            default: 'default.jpg',
        },

File: database/models/pharmacist.model.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        type: String,
+        required: [true, 'Email is required'],
+        unique: true,
+        lowercase: true,
+        trim: true
+    },
+    password: {
+        type: String,
+        required: [true, 'Password is required'],
+        minLength: [6, 'Password must be at least 6 characters long']
```
Comment (mennaseif) [inline database/models/pharmacist.model.ts]: add in password: select: false,

File: database/models/pharmacist.model.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    },
```
Comment (mennaseif) [inline database/models/pharmacist.model.ts:39]: passwordConfirm: {
            type: String,
            required: [true, 'Please confirm your password'],
            validate: {
                validator: function (el) {
                    return el === this.password;
                },
                message: 'Password are not the same!',
            },
        },

File: database/models/pharmacist.model.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        required: true
+    }
+}, { timestamps: true, versionKey: false });
+
+/* Create an index for the `licenseNumber` and `email` fields to improve query performance.
+ The index ensures that when queries involve searching for a pharmacist by `licenseNumber` or `email`,
+ MongoDB can quickly locate the relevant document(s) without scanning the entire collection.
+ The `1` signifies ascending order for the fields, which helps optimize the search process.
+*/
+pharmacistSchema.index({ licenseNumber: 1, email: 1 });
```
Comment (mennaseif) [inline database/models/pharmacist.model.ts]: مش مستاهله وبتقلل ال performance: index

File: database/models/pharmacist.model.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        type: Types.ObjectId,
+        ref: "Pharmacy",
+        required: true
+    },
+    inventory: {
+        type: Types.ObjectId,
+        ref: "inventory",
+        required: true
+    },
+    role: {
```
Comment (mennaseif) [inline database/models/pharmacist.model.ts]: role: in the top (first)

role: {
            type: String,
            default: 'pharmacist'
        }

File: database/models/pharmacist.model.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        type: String,
+        enum: ['admin', 'user'],
+        default: 'admin'
+    },
+    createdBy: {
```
Comment (mennaseif) [inline database/models/pharmacist.model.ts]: createdBy: delete it

File: database/models/pharmacist.model.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        minLength: [6, 'Password must be at least 6 characters long']
+    },
+    pharmacy: {
+        type: Types.ObjectId,
+        ref: "Pharmacy",
+        required: true
+    },
+    inventory: {
+        type: Types.ObjectId,
+        ref: "inventory",
```
Comment (mennaseif) [inline database/models/pharmacist.model.ts]: Inventory : capital I

File: database/models/pharmacist.model.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    },
+    role: {
+        type: String,
+        enum: ['admin', 'user'],
+        default: 'admin'
+    },
+    createdBy: {
+        type: Types.ObjectId,
+        ref: "User",  // Assuming you have a User model
+        required: true
```
Comment (mennaseif) [inline database/models/pharmacist.model.ts]: isBlocked:{
        type:Boolean,
        default: false,
    },

File: database/models/pharmacist.model.ts [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+        maxLength: 50
+    },
+    username: {
+        type: String,
+        required: [true, 'Username is required'],
+        unique: true,
+        minLength: [3, 'Username is too short'],
+        maxLength: 50
+    },
+    email: {
```
Comment (mennaseif) [inline database/models/pharmacist.model.ts:27]: validate with npm package 

validate: [validator.isEmail, 'Please provide a valid email'],
https://www.npmjs.com/package/validator

File: database/models/pharmacist.model.ts (hunk 4 above)
Comment (muhammad-nabih) [reply database/models/pharmacist.model.ts]: after all this I think a new pull request should be submitted What are all these comments?

**Comment by muhammad-nabih** [comment]
**Please Review My code Again**

File: database/models/pharmacist.model.ts [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+        },
+        message: "Passwords are not the same!",
+      },
+    },
+    pharmacy: {
+      type: Types.ObjectId,
+      ref: "Pharmacy",
+      required: true,
+    },
+    inventories: {
```
Comment (mennaseif) [inline database/models/pharmacist.model.ts]: Inventory onlyyyyyyyyyyyyyyyyyyyyyyyy without sssssssssssssssssssssssssss

**Review by mennaseif — CHANGES_REQUESTED** [review]

File: database/models/pharmacist.model.ts [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+    image: {
+      type: String,
+      default: "default.jpg",
+    },
+    email: {
+      type: String,
+      required: [true, "Email is required"],
+      unique: true,
+      lowercase: true,
+      trim: true,
```
Comment (mennaseif) [inline database/models/pharmacist.model.ts]: delete trim

**Review by mennaseif — CHANGES_REQUESTED** [review]

File: database/models/pharmacist.model.ts [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+      unique: true,
+      maxLength: 50,
+    },
+    username: {
+      type: String,
+      required: [true, "Username is required"],
+      unique: true,
+      minLength: [3, "Username is too short"],
+      maxLength: 50,
+    },
```
Comment (mennaseif) [inline database/models/pharmacist.model.ts:22]: add imageUrl:{
type:string
default:"default.jpg"
}

**Review by mennaseif — CHANGES_REQUESTED** [review]

File: database/models/pharmacist.model.ts (hunk 11 above)
Comment (mennaseif) [inline database/models/pharmacist.model.ts]: !!!

**Review by mennaseif — CHANGES_REQUESTED** [review]

File: database/models/pharmacist.model.ts (hunk 10 above)
Comment (muhammad-nabih) [reply database/models/pharmacist.model.ts]: okaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaay

File: database/models/pharmacist.model.ts (hunk 12 above)
Comment (muhammad-nabih) [reply database/models/pharmacist.model.ts:22]: done

**Review by mennaseif — APPROVED** [review]

**Pull request merged**